test/e2e/testkit/otlp/matchers/metric_matchers.go

```diff
@@ -34,6 +34,26 @@ func HaveMetrics(expectedMetrics ...pmetric.Metric) types.GomegaMatcher {
 	}, gomega.ContainElements(expectedMetrics))
 }
 
+func HaveNumberOfMetrics(expectedMetricCount int) types.GomegaMatcher {
+	return gomega.WithTransform(func(actual interface{}) (int, error) {
+		actualBytes, ok := actual.([]byte)
+		if !ok {
+			return 0, fmt.Errorf("HaveNumberOfMetrics requires a []byte, but got %T", actual)
+		}
+
+		actualMds, err := unmarshalOTLPJSONMetrics(actualBytes)
+		if err != nil {
+			return 0, fmt.Errorf("HaveNumberOfMetrics requires a valid OTLP JSON document: %v", err)
+		}
+		metricsCount := 0
+		for _, md := range actualMds {
+			metricsCount += len(metrics.AllMetrics(md))
+		}
+
+		return metricsCount, nil
+	}, gomega.Equal(expectedMetricCount))
+}
+
```

Comment:
For the rest of the matchers, we have some basic unit tests. I know that having tests for test helpers sounds like a bit of an overkill, but in this case, I think it's justified. It's not trivial and there is some JSON parsing involved. Could you please add a few?

Reply:
Done